### PR TITLE
Enable SMT array memory model

### PR DESCRIPTION
**Pull request by andreistefanescu**

### CODE REVIEW

File: src/SAWScript/Crucible/LLVM/Override.hs [hunk 1]
```diff
@@ -1226,16 +1229,47 @@ executePointsTo opts sc cc spec (LLVMPointsTo _loc ptr val) =
...
+     storTy <- Crucible.toStorableType memTy
+     val' <- liftIO $ instantiateSetupValue sc s val
+
+     smt_array_memory_model_enabled <- liftIO $ W4.getOpt
+      =<< W4.getOptionSetting
+        enableSMTArrayMemoryModel
+        (W4.getConfiguration sym)
+
+     mem'' <- case val' of
+      SetupTerm tm
```
Comment (siddharthist) [inline src/SAWScript/Crucible/LLVM/Override.hs]: This looks very similar to the code in `Builtins.LLVM`. Are they the same? Could we make a helper function for this?

**Comment by siddharthist** [comment]
Nit: this PR mixes snake case and camel case, which makes it just a tiny bit harder to remember what all the variables are named when editing the code

File: src/SAWScript/Crucible/LLVM/ResolveSetupValue.hs [hunk 2]
```diff
@@ -522,3 +539,116 @@ equalValsPred cc v1 v2 = go (v1, v2)
...
+
+        _ -> fail $ "unexpected cryptol type: " ++ show cryptol_type
+
+  case ttSchema typed_term of
+    Cryptol.Forall [] [] cryptol_type -> do
+      let evaluated_type = (Cryptol.evalValType Map.empty cryptol_type)
+      storage_size <- Crucible.storageTypeSize <$> toLLVMStorageType
+        data_layout
+        evaluated_type
+      padding_byte_term <- scBvConst saw_context 8 0xff
```
Comment (siddharthist) [inline src/SAWScript/Crucible/LLVM/ResolveSetupValue.hs]: Why this constant here? Does it just not matter what the padding bytes are set to?

Comment (andreistefanescu) [reply src/SAWScript/Crucible/LLVM/ResolveSetupValue.hs]: I just put something, the padding bytes are never accessed

**Comment by andreistefanescu** [comment]
@atomb I tested on the bike proofs, and it improved performance.

**Pull request merged**